lib/spack/spack/cmd/install.py

```diff
@@ -43,11 +45,13 @@
 description = "Build and install packages"
 
 
+
+
 def setup_parser(subparser):
     subparser.add_argument(
         '--only',
         default='package,dependencies',
-        dest='things_to_install',
+        dest='only_str',
```

Comment:
`only_str` is consistent with the argument named `only`.  Maybe this should be called `only` (except that `only` is really a set, but it's being encoded as a string)